app/controllers/api/KafkaStateCheck.scala

```diff
@@ -51,12 +51,18 @@ class KafkaStateCheck (val messagesApi: MessagesApi, val kafkaManagerContext: Ka
   }
 
   def unavailablePartitions(c: String, t: String) = Action.async { implicit request =>
-    kafkaManager.getTopicIdentity(c,t).map { errorOrTopicIdentity =>
+    kafkaManager.getTopicIdentity(c, t).map { errorOrTopicIdentity =>
       errorOrTopicIdentity.fold(
         error => BadRequest(Json.obj("msg" -> error.msg)),
-        topicIdentity => Ok(Json.obj("topic" -> t, "unavailablePartitions" -> topicIdentity.partitionsIdentity.filter(_._2.isr.isEmpty).map{case (num, pi) => pi.partNum}))
-      )
+        topicIdentity => Ok(Json.obj("topic" -> t, "unavailablePartitions" -> topicIdentity.partitionsIdentity.filter(_._2.isr.isEmpty).map { case (num, pi) => pi.partNum })))
     }
   }
-}
 
+  def topicSummary(cluster: String, consumer: String, topic: String) = Action.async { implicit request =>
+    kafkaManager.getConsumedTopicState(cluster, consumer, topic).map { errorOrTopicSummary =>
+      errorOrTopicSummary.fold(
+        error => BadRequest(Json.obj("msg" -> error.msg)),
+        topicSummary => Ok(Json.obj("totalLag" -> topicSummary.totalLag.get, "percentageCovered" -> topicSummary.percentageCovered)))
```

Comment:
topicSummary.totalLag.get will throw exception if the value is None
You should do a getOrElse to avoid that.
e.g. topicSummary.totalLag.getOrElse(Int.MinValue)